### PR TITLE
Develop 1.9 fwdport

### PR DESCRIPTION
**Pull request by SpacemanPaul**
### Reason for this pull request

Fold in recent changes from `develop` branch to `develop-1.9`.

 - [x] Tests added / passed
 - [x] Fully documented, including `docs/about/whats_new.rst` for all changes

### CODE REVIEW

File: datacube/index/abstract.py [hunk 1]
```diff
@@ -1141,30 +1141,32 @@ def archive(self, ids: Iterable[DSID]) -> None:
         :param Iterable[Union[str,UUID]] ids: list of dataset ids to archive
         """
 
-    def archive_less_mature(self, ds: Dataset) -> None:
+    def archive_less_mature(self, ds: Dataset, delta: int) -> None:
```
Comment (Ariana-B) [inline datacube/index/abstract.py]: Seems like this is missing the default value added in my latest PR

File: datacube/index/abstract.py [hunk 2]
```diff
@@ -1141,30 +1141,32 @@ def archive(self, ids: Iterable[DSID]) -> None:
...
-        less_mature = self.find_less_mature(ds)
+        less_mature = self.find_less_mature(ds, delta)
         less_mature_ids = map(lambda x: x.id, less_mature)
 
         self.archive(less_mature_ids)
         for lm_ds in less_mature_ids:
             _LOG.info(f"Archived less mature dataset: {lm_ds}")
 
-    def find_less_mature(self, ds: Dataset) -> Iterable[Dataset]:
+    def find_less_mature(self, ds: Dataset, delta: int) -> Iterable[Dataset]:
```
Comment (Ariana-B) [inline datacube/index/abstract.py]: Same as above

**Review by Ariana-B — CHANGES_REQUESTED** [review]
LGTM except for `archive_less_mature` and `find_less_mature` missing default delta value

File: datacube/index/abstract.py (hunk 1 above)
Comment (SpacemanPaul) [reply datacube/index/abstract.py]: D'oh!  Yes, I can see how that happened.  Thanks.

**Review by Ariana-B — APPROVED** [review]

**Pull request merged**